numpy/testing/utils.py

```diff
@@ -689,20 +689,13 @@ def chk_same_position(x_id, y_id, hasval='nan'):
 
         if isnumber(x) and isnumber(y):
             x_isnan, y_isnan = isnan(x), isnan(y)
-            x_isinf, y_isinf = isinf(x), isinf(y)
```

Comment:
So just for my edification, you are relying on the following being true then:

    +inf == +inf
    -inf == -inf
    -inf < +inf
    -inf < some_finite_number
    some_finite_number < +inf